cli/cmd/user/add.go

```diff
@@ -37,16 +39,44 @@ func AddCmd(ch *cmdutil.Helper) *cobra.Command {
 			}
 
 			if group != "" {
-				_, err := client.AddUsergroupMemberUser(cmd.Context(), &adminv1.AddUsergroupMemberUserRequest{
-					Organization: ch.Org,
-					Usergroup:    group,
-					Email:        email,
+				res, err := client.AddUsergroupMemberUser(cmd.Context(), &adminv1.AddUsergroupMemberUserRequest{
+					Organization:     ch.Org,
+					Usergroup:        group,
+					Email:            email,
+					AddRillUserToOrg: false,
 				})
 				if err != nil {
+					if s, ok := status.FromError(err); ok && s.Code() == codes.FailedPrecondition && strings.Contains(s.Message(), "user is not a member of the org") {
+						// ask if user should be added to the organization
+						addToOrg, err := cmdutil.ConfirmPrompt(fmt.Sprintf("This user exists in Rill but not part of the %q organization. Do you want to add the user to the organization as viewer?", ch.Org), "", false)
+						if err != nil {
+							return err
+						}
+
+						if addToOrg {
+							_, err = client.AddUsergroupMemberUser(cmd.Context(), &adminv1.AddUsergroupMemberUserRequest{
+								Organization:     ch.Org,
+								Usergroup:        group,
+								Email:            email,
+								AddRillUserToOrg: true,
```

Comment:
You could also keep it simple by removing this flag, and just calling `AddOrganizationMemberUser` separately first. That would mean you don't need to handle the invite creation logic in multiple RPCs (and wouldn't need a `tx` in this RPC).